File: devops-scm-api/src/main/java/com/tencent/devops/scm/api/pojo/auth/TokenSshPrivateKeyScmAuth.java
package com.tencent.devops.scm.api.pojo.auth;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class TokenSshPrivateKeyScmAuth implements IScmAuth {
    public static final String CLASS_TYPE = "TOKEN_SSH_PRIVATEKEY";

    private String token;
    private String privateKey;
    private String passphrase;
}

File: devops-scm-api/src/main/kotlin/com/tencent/devops/scm/api/pojo/auth/TokenSshPrivateKeyScmAuth.kt
package com.tencent.devops.scm.api.pojo.auth

class TokenSshPrivateKeyScmAuth(
    val token: String,
    val privateKey: String,
    val passphrase: String? = null
) : IScmAuth {
    companion object {
        const val CLASS_TYPE: String = "TOKEN_SSH_PRIVATEKEY"
    }
}
